### PR TITLE
Changes for API publish to bintray and github . Also versioning API change.

### PR DESCRIPTION
**Pull request by prashanthlgowda**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -101,7 +130,7 @@ dependencies {
 	compile 'com.dell.isg.smi:adapter-chassis:1.0.29'
     compile 'com.dell.isg.smi:commons-elm:1.0.82'
     compile 'com.dell.isg.smi:commons-utilities:1.0.32'
-	compile 'com.dell.isg.smi:commons-model:1.0.89'
+	compile 'com.dell.isg.smi:commons-model:1.0-SNAPSHOT'
```
Comment (michaelsteven) [inline build.gradle]: Change to 1.0.97

File: src/docs/asciidoc/examples.adoc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+Based on the the device type identified , the summary response schema could be any one of the following..
+
+Supported - SERVER,CHASSIS and STORAGE (Complellent) .Not supported SWITCH. 
+
+SERVER
+~~~~~~
+
+Types -  IDRAC7, IDRAC8 and IDRAC9
+
+Example fortThe schema of summary response:
```
Comment (michaelsteven) [inline src/docs/asciidoc/examples.adoc]: fix typo

File: src/docs/asciidoc/examples.adoc [hunk 3]
```diff
@@ -0,0 +1,165 @@
+ Device Summary Response
+------------------------
+
+Based on the the device type identified , the summary response schema could be any one of the following..
```
Comment (michaelsteven) [inline src/docs/asciidoc/examples.adoc]: remove repeated "the" and space before coma.  Add a colon instead of ..

File: gradle.properties [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 dockerTag=devel
-artifactory_contextUrl=https://gtie-artifactory.us.dell.com/artifactory
 dependency_repo=libs-snapshot
 buildInfo.licenseControl.runChecks=true
 buildInfo.licenseControl.autoDiscover=true
 buildInfo.build.name=com.dell.isg.smi:service-device-discovery
-systemProp.sonar.host.url=http://100.68.126.201:9000
\ No newline at end of file
+bintrayRepo=smi-service-device-discovery
+bintrayUserOrg=rackhd-smi
```
Comment (michaelsteven) [inline gradle.properties]: change org to rackhd

File: gradle.properties [hunk 5]
```diff
@@ -1,8 +1,9 @@
 version=1.0
 dockerTag=devel
-artifactory_contextUrl=https://gtie-artifactory.us.dell.com/artifactory
 dependency_repo=libs-snapshot
 buildInfo.licenseControl.runChecks=true
 buildInfo.licenseControl.autoDiscover=true
 buildInfo.build.name=com.dell.isg.smi:service-device-discovery
-systemProp.sonar.host.url=http://100.68.126.201:9000
\ No newline at end of file
+bintrayRepo=smi-service-device-discovery
```
Comment (michaelsteven) [inline gradle.properties]: change to docs/apidoc

File: gradle.properties [hunk 6]
```diff
@@ -1,8 +1,9 @@
...
+bintrayUserOrg=rackhd-smi
+bintrayVcsUrl=https://rackhd-smi.bintray.com/smi-service-device-discovery
```
Comment (michaelsteven) [inline gradle.properties]: change to "https://bintray.com/rackhd/docs/apidoc"

File: build.gradle (hunk 1 above)
Comment (prashanthlgowda) [reply build.gradle]: Done

File: gradle.properties (hunk 5 above)
Comment (prashanthlgowda) [reply gradle.properties]: done

File: gradle.properties (hunk 4 above)
Comment (prashanthlgowda) [reply gradle.properties]: done

File: src/docs/asciidoc/examples.adoc (hunk 2 above)
Comment (prashanthlgowda) [reply src/docs/asciidoc/examples.adoc]: done

File: src/docs/asciidoc/examples.adoc (hunk 3 above)
Comment (prashanthlgowda) [reply src/docs/asciidoc/examples.adoc]: done

File: gradle.properties (hunk 6 above)
Comment (prashanthlgowda) [reply gradle.properties]: done

File: src/docs/asciidoc/readme.adoc [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+Example Payload 1.
+
+Use global credentials (override all credentials in application yml for all devices) 
+....
+/api/1.0/discover/range
+
+{
+   "credential":{
+      "user":"root",
```
Comment (phelpdh) [inline src/docs/asciidoc/readme.adoc]: Change to userName

File: src/docs/asciidoc/readme.adoc [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+Example Payload 4.
+
+The example payload below will attempt to discover servers and chassis in the ip range 100.68.124.11 - 57, using the global credentials "root" and "calvin". It will also attempt to discover switches in the ip range of 100.68.124.6 - 10, using the range specific credentials "abc0" and "xyz0"
+
+....
+
+/api/1.0/discover/range
+{
+   "credential":{
+      "user": "root",
```
Comment (phelpdh) [inline src/docs/asciidoc/readme.adoc]: Change to userName

File: src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java [hunk 9]
```diff
@@ -36,9 +35,8 @@
 
     private static final Logger logger = LoggerFactory.getLogger(DeviceDiscoveryController.class.getName());
 
-
     @RequestMapping(value = "/range", method = RequestMethod.POST, headers = "Accept=application/json", consumes = "application/json", produces = "application/json")
-    @ApiOperation(value = "/range", nickname = "range", notes = "This operation allow user to collect server software identity throu wsman.", response = DiscoverdDeviceResponse.class, responseContainer = "List")
+    @ApiOperation(value = "/range", nickname = "range", notes = "This operation ping sweep and discovers the devices within the given range of IP.", response = DiscoverdDeviceResponse.class, responseContainer = "List")
```
Comment (phelpdh) [inline src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java]: Change to :
This operation will ping sweep and discover devices within the given IP range.

File: src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java [hunk 10]
```diff
@@ -55,7 +53,7 @@
 
 
     @RequestMapping(value = "/ips", method = RequestMethod.POST, headers = "Accept=application/json", consumes = "application/json", produces = "application/json")
-    @ApiOperation(value = "/ips", nickname = "ips", notes = "This operation allow user to collect server software identity throu wsman.", response = DiscoverdDeviceResponse.class, responseContainer = "List")
+    @ApiOperation(value = "/ips", nickname = "ips", notes = "This operation ping sweep and discovers the devices within the given list of IP.", response = DiscoverdDeviceResponse.class, responseContainer = "List")
```
Comment (phelpdh) [inline src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java]: Change to:
This operation will ping sweep and discover devices within the given IP list.

File: src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java (hunk 9 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java]: Done

File: src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java (hunk 10 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/service/device/discovery/controller/DeviceDiscoveryController.java]: Done

File: src/docs/asciidoc/readme.adoc (hunk 8 above)
Comment (prashanthlgowda) [reply src/docs/asciidoc/readme.adoc]: Done

File: src/docs/asciidoc/readme.adoc (hunk 7 above)
Comment (prashanthlgowda) [reply src/docs/asciidoc/readme.adoc]: Done

**Pull request merged**